src/envoy/http/path_matcher/filter_test.cc

```diff
@@ -166,7 +166,7 @@ TEST_F(PathMatcherFilterTest, DecodeHeadersNoMatch) {
   EXPECT_CALL(
       mock_cb_,
       sendLocalReply(Envoy::Http::Code::NotFound,
-                     "`POST /bar` does not match any requirement URI template.",
+                     "`POST /bar` does not match any defined routes.",
```

Comment:
"defined routes" is confusing too.  When users see this?  what can they do?

I think we can mention "service config".  For now, consumer client (or users) are the same as producer, they know about "service config"

Reply:
Well, not really. For Cloud API Gateway, API Producers don't know about service config either. They only know about OpenAPI spec, service config should be transparent to them.

Reply:
how about
.... is not defined in service spec.  

OpenAPI is a service spec. 
It will also applies to gRPC too,  grpc spec is another service spec.

Reply:
Make sense.